index.html

```diff
@@ -226,14 +226,14 @@ <h2>The <dfn>WoT</dfn> object</h2>
       // [SecureContext]
       // [NamespaceObject]
       interface WoT {
-        Observable&lt;ThingDescription&gt; discover(optional ThingFilter filter);
+        Observable discover(optional ThingFilter filter);
         Promise&lt;ThingDescription&gt; fetch(USVString url);
         ConsumedThing consume(ThingDescription td);
         ExposedThing produce(ThingModel model);
         Promise&lt;void&gt; register(USVString directory, ExposedThing thing);
         Promise&lt;void&gt; unregister(USVString directory, ExposedThing thing);
       };
-      typedef dictionary ThingTemplate;
+      typedef object ThingTemplate;
```

Comment:
A ConsumedThing instance can serve as / implements ThingFragment. It has all the required fields and instance-specific fields must be avoided anyway.

Reply:
I wasn't able to figure out the single line typedefs either. 
ThingModel could be a ConsumedThing object, i.e. you consume a thing for some reason and reuse the object (rather than fetching again the TD) to create an ExposedThing based on that.

Reply:
And as said, as `produce()` takes `ThingFragment`, you can pass a `ConsumedThing` there, because it implements (or extends) `ThingFragment`. Hence, it is fine when `ThingModel` is `ThingDescription | ThingFragment`.